api/kv_v2.go

```diff
@@ -417,7 +502,7 @@ func (kv *kvv2) Rollback(ctx context.Context, secretPath string, toVersion int)
 
 func extractCustomMetadata(secret *Secret) (map[string]interface{}, error) {
 	// Logical Writes return the metadata directly, Reads return it nested inside the "metadata" key
-	cmI, ok := secret.Data["custom_metadata"]
+	cmI, ok := secret.Data[customMetadataKey]
```

Comment:
Same comment as above regarding variable names. It's hard to tell what `cmI` stands for